app/scripts/modules/core/src/pipeline/service/execution.service.ts

```diff
@@ -43,8 +47,23 @@ export class ExecutionService {
         });
     }
 
-    public getExecutions(applicationName: string): IPromise<IExecution[]> {
-      return this.getFilteredExecutions(applicationName);
+  /**
+   * Returns a filtered list of executions for the given application
+   * @param {string} applicationName the name of the application
+   * @param {Application} application: if supplied, and pipeline parameters are present on the filter model, the
+   * application will be used to correlate and filter the retrieved executions to only include those pipelines
+   * @return {<IExecution[]>}
+   */
+    public getExecutions(applicationName: string, application: Application = null): IPromise<IExecution[]> {
```

Comment:
I don't love that the presence of the application argument triggers slightly different behavior (additional filtering), but not sure there's a clean way to do it without refactoring a bunch of other stuff.